packages/table/preview/index.tsx

```diff
@@ -468,6 +528,31 @@ class MutableTable extends React.Component<{}, IMutableTableState> {
     // State updates
     // =============
 
+    // designed to be called from componentWillMount and componentWillUpdate, hence it expects nextProps
+    private syncCellContent = (nextState?: IMutableTableState) => {
+        const isNextStateDefined = nextState != null;
+
+        if (isNextStateDefined &&
+            nextState.cellContent === this.state.cellContent
+            && nextState.numRows === this.state.numRows
+            && nextState.numCols === this.state.numCols
+        ) {
+            return;
+        }
+
+        const cellContent = isNextStateDefined ? nextState.cellContent : this.state.cellContent;
+        const numRows = isNextStateDefined ? nextState.numRows : this.state.numRows;
+        const numCols = isNextStateDefined ? nextState.numCols : this.state.numCols;
```

Comment:
👎  either use `this.state` as default value for argument, or deconstruct all at once like so:

```ts
const { ... } = isNextStateDefined ? nextState : this.state;
```